mrf/era-1/basic-syntax-#9.js

```diff
@@ -0,0 +1,189 @@
+  let result = Math.random();
+
+  if (result >= 0.5) return "HEAD";
+  else return "TAIL";
+}
+
+//! test flipACoin func
+console.log("flipACoin:", flipACoin());
+
+// concat two strings
+function concat(stringA, stringB) {
+  // return stringA + stringB;
+  return stringA.concat(stringB);
+}
+
+//! Test concat func
+console.log("concat: ", concat("abc", "def"));
+
+// 2^x = ?
+function power2(x) {
+  // return Math.pow(2, x);
+  return 2 ** x;
+}
+
+//! test power2 func
+console.log("power2 of 10:", power2(10));
+
+// user object with first name and last name, both capitalized
+// e.g. getUser('hello', 'world') is { firstName: 'Hello', lastName: 'World' }
+function getUser(firstName, lastName) {
+  firstName = firstName.charAt(0).toUpperCase() + firstName.slice(1);
+  lastName = lastName.charAt(0).toUpperCase() + lastName.slice(1);
+  const arr = { firstName: firstName, lastName: lastName };
+  return arr;
+}
+
+//! test getUser function
+console.log("getUser whami:", getUser("mmd", "mmdi"));
+
+// greet someone!
+// e.g. greet('ConnecMent') is "Hello, ConnecMent! Hope you the best."
+function greet(name) {
+  // return "Hello, " + name + " Hope you the best.";
+  return `Hello, ${name} Hope you the best.`;
+}
+
+//! test greet func
+console.log("Greet:", greet("Mr MRF"));
+
+// get array of n random numbers
+function getNRandomNumbers(n) {
+  const final = new Array(n).fill();
+
+  // for (let i = 0; i < n; i++) {
+  //   final[i] = Math.random();
+  // }
+  // return final;
+
+  // or
+  return final.map((x) => (x = Math.random()));
+}
+
+//! test of getNRandomNumbers
+console.log("getNRandomNumbers: 10", getNRandomNumbers(10));
+
+// get sorted array of n random numbers, in range min to max
+function getNRandomNumbersRange(n, min, max) {
+  const final = new Array();
+
+  for (let i = 0; i < n; i++) {
+    // plus 1 to included max
+    let x = Math.floor(Math.random() * (max - min + 1)) + min;
+    final.push(x);
+  }
+
+  return final;
+}
+
+//! test of getNRandomNumbersRange
+console.log(
+  "getNRandomNumbersRange: 10, 5, 15",
+  getNRandomNumbersRange(10, 5, 15)
+);
+
+// log current time with a x seconds interval, for n seconds
+function logCurrentTime(x, n) {
+  function printTime() {
+    let date = new Date();
+    console.log(
+      "Time:",
+      date.getHours(),
+      ":",
+      date.getMinutes(),
+      ":",
+      date.getSeconds()
+    );
+  }
+
+  // setTimeout(printTime, x * 1000);
+
+  for (let i = 0; i < n / x; i++) {
+    printTime();
+
+    let time1 = new Date().getTime() / 1000;
+    let time2 = new Date().getTime() / 1000;
+
+    while (time1 + x > time2) {
+      time2 = new Date().getTime() / 1000;
+    }
+  }
+}
+
+//! test of logCurrentTime
+logCurrentTime(2, 10);
+
+// higher order function: call the function in the parameters list
+// e.g. callFuncNTimes(flipACoin, 3) calls flipACoin 3 times
+function callFuncNTimes(func, n, ...funcParam) {
+  for (let i = 0; i < n; i++) {
+    func(funcParam);
```

Comment:
You need to spread the `funcParam`:
```suggestion
    func(...funcParam);
```

Reply:
got it 🌾